mava/utils/logger.py

```diff
@@ -41,19 +45,73 @@ class LogEvent(Enum):
     MISC = "misc"
 
 
-class MavaLogger:
-    """The main logger for Mava systems.
+def winrate_custom_metric(metrics: Dict, config: DictConfig) -> Dict:
+    """Calculate win rate from episode metrics.
+
+    Args:
+    ----
+        metrics: Dictionary containing 'won_episode' and 'is_terminal_step'.
+        config: The system config.
 
-    Thin wrapper around the MultiLogger that is able to describe arrays of metrics
-    and calculate environment specific metrics if required (e.g winrate).
+    Returns:
+    -------
+        Dictionary with added 'win_rate' metric and 'won_episode' removed.
     """
+    if "won_episode" not in metrics:
+        return metrics
+
+    # Count the number of terminal steps to determine episode count
+    is_terminal_steps = metrics.get("is_terminal_step", np.array([]))
+    n_episodes: int = np.sum(is_terminal_steps)
+
+    # If no episodes were completed, return unchanged metrics
+    if n_episodes == 0:
+        return metrics
+
+    # Calculate the win rate
+    n_won_episodes: int = np.sum(metrics["won_episode"])
+    win_rate: float = (n_won_episodes / n_episodes) * 100
+
+    # Update metrics
+    metrics["win_rate"] = win_rate
+    metrics.pop("won_episode")
+
+    return metrics
+
 
-    def __init__(self, config: DictConfig) -> None:
+class MavaLogger:
+    def __init__(
+        self,
+        config: DictConfig,
+        custom_metrics_fn: Callable[[Dict, DictConfig], Dict] = winrate_custom_metric,
```

Comment:
Doesn't this put us in a bit of tricky situation moving forward if environments have many custom metrics since this looks like it assumes we will only have one custom metric at a time?

Or if the custom metric need to extra processing like win rate does, it should just log those with no problem?

Reply:
So if you had more custom metrics that need processing you just create a function that handles all of them, the function just takes all the metrics so it would modify the ones that it cares about. 

But that's only for custom metrics that need to be modified many metrics are probably fine just as is

Reply:
It's sort of only a win, we get the option to log these metrics where as before it was tricky and you had to modify the logger itself and was a bit hacky to pass metrics all the way through to the logger. So if envs have custom metrics and we don't want to support them then we just don't add them to `extras["env_metrics"]` to begin with, otherwise we add them and process them if need be.

Reply:
Perfect, thanks for clarifying. Can you make just add a comment in the logger that briefly explains this?